cmd/sriovdp/sriov-device-plugin.go

```diff
@@ -149,41 +181,49 @@ func (sm *sriovManager) discoverNetworks() error {
 			}
 			glog.Infof("Number of Configured VFs for device %v is %v", dev, string(configuredVFs))
 
-			//get PCI IDs for VFs
-			for vf := 0; vf < numconfiguredvfs; vf++ {
-				vfDir := fmt.Sprintf("/sys/class/net/%s/device/virtfn%d", dev, vf)
-				dirInfo, err := os.Lstat(vfDir)
-				if err != nil {
-					glog.Errorf("Error. Could not get directory information for device: %s, VF: %v. Err: %v", dev, vf, err)
-					return err
-				}
-
-				if (dirInfo.Mode() & os.ModeSymlink) == 0 {
-					glog.Errorf("Error. No symbolic link between virtual function and PCI - Device: %s, VF: %v", dev, vf)
-					return fmt.Errorf("Error. No symbolic link between virtual function and PCI - Device: %s, VF: %v", dev, vf)
-				}
-
-				pciInfo, err := os.Readlink(vfDir)
-				if err != nil {
-					glog.Errorf("Error. Cannot read symbolic link between virtual function and PCI - Device: %s, VF: %v. Err: %v", dev, vf, err)
-					return err
-				}
-
-				pciAddr := pciInfo[len("../"):]
-				glog.Infof("PCI Address for device %s, VF %v is %s", dev, vf, pciAddr)
-
-				devName := pciAddr
-				sm.devices[devName] = pluginapi.Device{ID: devName, Health: pluginapi.Healthy}
```

Comment:
I see you have refactored this section and extracted the GetVFList() method. But I don't see the initial device list is being populated anywhere. We only get a list of device only after a Probe() is invoked. The ListAndWatch() should send the the initial device list(which is now empty) first and will send updated list only when any device's health is changed.

Reply:
The ListAndWatch() updates device information by calling Probe() first; when sm.devices[] list is empty, Probe() assigns default value to it and set the 'changed' flag to true which triggers the ListAndWatch to send ListAndWatchResponse back. The initial assignment of sm.devices[] is at https://github.com/intel/sriov-network-device-plugin/pull/32/files#diff-8dbc0a94295122ac9cb2460632637cbdR220

Reply:
Ok, thanks for explanation.